src/lib/translation.ts

```diff
@@ -29,14 +29,15 @@ export async function getTranslation(string: string, options: TranslationOptions
     return response.text.join(' ')
   }
 
-  if (options.translationService === TranslationService.deepl) {
+  if (options.translationService === TranslationService.deepl || options.translationService === TranslationService.deeplFree) {
     params.set('auth_key', options.apiKey)
     params.set('target_lang', options.toLocale)
     params.set('tag_handling', 'xml')
     params.set('text', string)
 
+    const endpoint = options.translationService === TranslationService.deepl ? 'https://api.deepl.com': 'https://api-free.deepl.com'
```

Comment:
You're only interested in the `api` or `api-free` part of the endpoint url. So, using the `isDeepL` and `isDeepLFree` constants, you could write this as follows:
```
const apiVersion = isDeepLFree ? 'api-free' : 'api'
const request = await fetch(
    `https://${apiVersion}.deepl.com/v2/translate?${params.toString()}`,
)
```